lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -668,68 +753,127 @@ class MongoClientInterface {
         });
     }
 
+    /**
+     * puts object metadata in bucket
+     * @param {String} bucketName bucket name
+     * @param {String} objName object name
+     * @param {Object} objVal object metadata
+     * @param {object} params params
+     * @param {Object} log logger
+     * @param {Function} cb callback
+     * @return {undefined}
+     */
     putObject(bucketName, objName, objVal, params, log, cb) {
         MongoUtils.serialize(objVal);
         const c = this.getCollection(bucketName);
-        if (params && params.versioning && !params.versionId) {
-            return this.putObjectVerCase1(c, bucketName, objName, objVal,
-                params, log, cb);
-        } else if (params && params.versionId === '') {
-            return this.putObjectVerCase2(c, bucketName, objName, objVal,
-                params, log, cb);
-        } else if (params && params.versionId && !params.repairMaster) {
-            return this.putObjectVerCase3(c, bucketName, objName, objVal,
-                params, log, cb);
-        } else if (params && params.versionId && params.repairMaster) {
-            return this.putObjectVerCase4(c, bucketName, objName, objVal,
+        return this.getBucketVFormat(bucketName, log, (err, vFormat) => {
+            if (err) {
+                return cb(err);
+            }
+            if (!params) {
```

Comment:
I'd suggest creating a new object `_params = Object.assign({}, params)` to avoid modifying the argument, since it's an API function we cannot assume that it's safe to modify `params`.